test/LANDRegistry.js

```diff
@@ -1098,4 +1097,150 @@ contract('LANDRegistry', accounts => {
+    })
+
+    it('should return true if decayed LAND', async () => {
+      const assetId = await land.encodeTokenId(0, 1)
+      await land.setGracePeriod(1, sentByCreator)
+      await land.setDeemPeriod(2, sentByCreator)
+      const decayed = await land.hasDecayed(assetId)
+      decayed.should.be.true
+    })
+
+    it('should return false if no gracePeriod set', async () => {
+      const assetId = await land.encodeTokenId(0, 1)
+      const decayed = await land.hasDecayed(assetId)
+      decayed.should.be.false
+    })
+
+    it('should return false if gracePeriod is bigger than today', async () => {
+      const assetId = await land.encodeTokenId(0, 1)
+      const now = new Date().getMilliseconds() / 1000
+      await land.setGracePeriod(now, sentByCreator)
+      const decayed = await land.hasDecayed(assetId)
+      decayed.should.be.false
+    })
+
+    it('should return false if not decayed LAND', async () => {
+      const assetId = await land.encodeTokenId(0, 1)
+      const now = new Date().getMilliseconds() * 1000
+      await land.setGracePeriod(1, sentByCreator)
+      await land.setDeemPeriod(now * now, sentByCreator)
+      const decayed = await land.hasDecayed(assetId)
+      decayed.should.be.false
+    })
+
+    it('should reactivate decayed LAND and emit reactivate event', async () => {
+      await land.setGracePeriod(1, sentByCreator)
+      await land.setDeemPeriod(2, sentByCreator)
+      let owner = await land.ownerOfLand(0, 1)
+      owner.should.be.equal(user)
+      await land.setReactivateAuthorized(operator, true, sentByCreator)
+      const assetId = await land.encodeTokenId(0, 1)
+      const { logs } = await land.reactivate(
+        assetId,
+        anotherUser,
+        sentByOperator
+      )
+
+      // Event
+      const log = logs[1]
+      log.event.should.be.eq('Reactivate')
+      log.args._assetId.should.be.bignumber.equal(assetId)
+      log.args._newOwner.should.be.equal(anotherUser)
+      // TODO uncomment when reactivate implemented
+      // owner = await land.ownerOfLand(0, 1)
+      // owner.should.be.equal(anotherUser)
+    })
+
+    it('should refresh latestPing if reactivated LAND', async () => {
+      const assetId = await land.encodeTokenId(0, 1)
+      const latestPingBefore = await land.latestPing(anotherUser)
+      await land.setGracePeriod(1, sentByCreator)
+      await land.setDeemPeriod(2, sentByCreator)
+      await land.setReactivateAuthorized(operator, true, sentByCreator)
+      await land.reactivate(assetId, anotherUser, sentByOperator)
+      const latestPingAfter = await land.latestPing(anotherUser)
+      ;(latestPingBefore < latestPingAfter).should.be.true
+    })
+
+    it('should refresh latestPing if pinged by owner', async () => {
+      const latestPingBefore = await land.latestPing(user)
+      await land.pingMyself(sentByUser)
+      const latestPingAfter = await land.latestPing(user)
+      ;(latestPingBefore < latestPingAfter).should.be.true
```

Comment:
What do you think of using a variable to store it? It will make the test more readable.

Also, it would be great if you can include the event emission check.

Reply:
Please, try checking against the specific time of the ping, it will make the test solid